src/Functions/Security/Get-RsCatalogItemRole.ps1

```diff
@@ -0,0 +1,117 @@
+            Use the "Connect-RsReportServer" function to set/update a default value.
+        
+        .PARAMETER Proxy
+            Report server proxy to use.
+            Use "New-RsWebServiceProxy" to generate a proxy object for reuse.
+            Useful when repeatedly having to connect to multiple different Report Server.
+        
+        .EXAMPLE
+            Get-RsCatalogItemRole -Identity 'jmcgee'
+            Description
+            -----------
+            This command will establish a connection to the Report Server located at http://localhost/reportserver using current user's credentials and then retrieves all access for user 'jmcgee'.
+        
+        .EXAMPLE
+            Get-RsCatalogItemRole -ReportServerUri 'http://localhost/reportserver_sql2012' -Identity 'jmcgee'
+            Description
+            -----------
+            This command will establish a connection to the Report Server located at http://localhost/reportserver_2012 using current user's credentials and then retrieves all access for user 'jmcgee'.
+
+        .EXAMPLE
+            Get-RsCatalogItemRole -ReportServerUri 'http://localhost/reportserver_sql2012' -path '/Path/Human Resources/'
+            Description
+            -----------
+            This command will establish a connection to the Report Server located at http://localhost/reportserver_2012 using current user's credentials and then retrieves all access on catalog items found at '/Path/Human Resources/'.
+        
+        .EXAMPLE
+            Get-RsCatalogItemRole -Credential 'CaptainAwesome' -Identity 'jmcgee' -Recurse
+            Description
+            -----------
+            This command will establish a connection to the Report Server located at http://localhost/reportserver using CaptainAwesome's credentials and then retrieves all access for user 'jmcgee' recursively.
+    #>
+
+    [CmdletBinding()]
+    param
+    (
+        [Alias('UserOrGroupName')]
+        [string]
+        $Identity,
+        
+        [Alias('ItemPath')]
+        [string]
+        $Path = "/",
+
+        [switch]
+        $Recurse,
+        
+        [string]
+        $ReportServerUri,
+        
+        [Alias('ReportServerCredentials')]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+        
+        $Proxy
+    )
+    
+    Begin
+    {
+        $Proxy = New-RsWebServiceProxyHelper -BoundParameters $PSBoundParameters
+    }
+        
+    Process
+    {
+        $Items = $Proxy.ListChildren($Path, $Recurse) 
+        $InheritParent = $true
+
+        foreach($Item in $Items)
+        {
+            foreach($Policy in $Proxy.GetPolicies($Item.path, [ref]$InheritParent))
+            {
+                # Drop Users/Groups that do not mach the Filter
+                if($Identity) {
+                    if($Policy.GroupUserName -ne $Identity) {
+                        Continue
```

Comment:
I suggest to do this before the foreach with a where-object to filter the results of getpolicies
something like

$Policies = $Proxy.GetPolicies($Item.path, [ref]$InheritParent) | Where-Object { $_.GroupUserName -eq $Identity }